yt_dlp/extractor/mastodon.py

```diff
@@ -0,0 +1,515 @@
+        }
+
+
+class MastodonIE(MastodonBaseIE):
+    # TODO: Add md5s
+    _TESTS = [{
+        'note': 'short form, compatible with haruhi-dl\'s usage',
+        'url': 'mastodon:mstdn.jp:105395495018076252',
+        'info_dict': {
+            'id': '105395495018076252@mstdn.jp',
+            'ext': 'mp4',
+            'uploader_id': 'nao20010128nao',
+            'like_count': int,
+            'comment_count': int,
+            'duration': 131.598,
+            'thumbnail': 'https://media.mstdn.jp/media_attachments/files/033/830/003/small/e8429d6ee1013c3e.png',
+            'title': 'てすや\nhttps://www.youtube.com/watch?v=jx0fBBkaF1w',
+            'uploader': 'Lesmiscore',
+            'uploader_url': 'https://mstdn.jp/@nao20010128nao',
+            'repost_count': int,
+        },
+    }, {
+        # Soapbox, audio file
+        'url': 'mastadon:https://gleasonator.com/notice/9zvJY6h7jJzwopKAIi',
+        'info_dict': {
+            'id': '9zvJY6h7jJzwopKAIi@gleasonator.com',
+            'title': '#FEDIBLOCK',
+            'ext': 'oga',
+            'comment_count': int,
+            'repost_count': int,
+            'uploader_url': 'https://gleasonator.com/users/alex',
+            'uploader': 'Alex Gleason',
+            'uploader_id': 'alex',
+            'like_count': int,
+        },
+    }, {
+        # gab social
+        'url': 'https://gab.com/ACT1TV/posts/104450493441154721',
+        'info_dict': {
+            'id': '104450493441154721@gab.com',
+            'ext': 'mp4',
+            'comment_count': int,
+            'uploader': 'Bill Blaze',
+            'uploader_url': 'https://gab.com/ACT1TV',
+            'uploader_id': 'ACT1TV',
+            'repost_count': int,
+            'duration': 12.422667,
+            'thumbnail': 'https://media.gab.com/system/media_attachments/files/056/821/863/small/0b5645cee32909e9.png',
+            'title': 'He shoots, he scores and the crowd went wild.... #Animal #Sports',
+            'like_count': int,
+        },
+    }, {
+        'url': 'https://gab.com/SomeBitchIKnow/posts/107163961867310434',
+        'md5': '8ca34fb00f1e1033b5c5988d79ec531d',
+        'info_dict': {
+            'id': '107163961867310434@gab.com',
+            'ext': 'mp4',
+            'comment_count': int,
+            'repost_count': int,
+            'title': 'md5:204055fafd5e1a519f5d6db953567ca3',
+            'uploader_id': 'SomeBitchIKnow',
+            'duration': 104.213333,
+            'uploader_url': 'https://gab.com/SomeBitchIKnow',
+            'like_count': int,
+            'uploader': 'L',
+            'thumbnail': 'https://media.gab.com/system/media_attachments/files/088/718/344/small/fded4ef06989f1bb.png',
+        }
+    }, {  # FIXME
+        'url': 'https://gab.com/TheLonelyProud/posts/107045884469287653',
+        'md5': 'f9cefcfdff6418e392611a828d47839d',
+        'info_dict': {
+            'id': '107045884469287653@gab.com',
+            'ext': 'mp4',
+        }
+    }, {
```

Comment:
Bug? @Lesmiscore

Reply:
wtf is this... I don't remember
https://github.com/yt-dlp/yt-dlp/blob/573a98d6f0867f9acb909cb3ff3dc9c10f9b2e8b/yt_dlp/extractor/gab.py#L85-L97